plugins/rbac/src/components/ConditionalAccess/CustomArrayField.tsx

```diff
@@ -35,8 +35,9 @@ export const CustomArrayField = (props: FieldProps) => {
         label={name}
         value={fieldVal}
         onChange={e => {
-          setFieldVal(e.target.value);
-          onChange(e.target.value.split(',').map(val => val.trim()));
+          const value = e.target.value;
+          setFieldVal(value);
+          onChange(value ? value.split(',').map(val => val.trim()) : []);
```

Comment:
So if a `minItems` attribute is available in the paramsSchema of the array param then an empty array will return a validation error. But when the array param for example kind is just listed as required this will not return any error as we are returning [] and its an acceptable value. Looks good to me.

Reply:
If it returns an empty array, the item account will be 0 which doesn't meet the `minItem: 1` requirement and it throws an error.

Reply:
@AndrienkoAleksandr I am able to add below condition on catalog-entity :
```
        {
            "rule": "IS_ENTITY_KIND",
            "resourceType": "catalog-entity",
            "params": {
                "kinds": []
            }
         }
```
On applying above the corresponding role members cannot access any type of catalog-entity kinds. Does this looks correct to you?

Reply:
We can deny such conditions with help of validator in the frame of the  https://issues.redhat.com/browse/RHIDP-2157 . But maybe it can be feature...

Reply:
But here https://docs.google.com/document/d/1XMEmSlwVog1cOD0X5ADwg53Vnm8PL-HH_8LZriUM7vs/edit?disco=AAABNnL7BR8 kinds is just required which mean kinds cannot be undefined but it can be [] also it doesnt have `minItems: 1` to restrict an empty array. So can this might be what plugin conidition rules author expects or its a backend bug?

Reply:
@AndrienkoAleksandr ^^

Reply:
> So can this might be what plugin condition rules author expects or its a backend bug?

I believe we are not responsible for such arrays. The plugin author designs and controls this aspect.